src/inmanta/export.py

```diff
@@ -189,8 +190,36 @@ def _load_resources(self, types: Dict[str, Entity]) -> None:
                             instance.location,
                         )
 
+        self._load_resource_sets(types, resource_mapping)
         Resource.convert_requires(resource_mapping, ignored_set)
 
+    def _load_resource_sets(self, types: Dict[str, Entity], resource_mapping: Dict["Instance", "Resource"]) -> None:
+        """
+        load the resource_sets in a dict with as keys resource_ids and as values the name of the resource_set
+        the resource belongs to.
+        This method should only be called at the end of self._load_resources()
+        """
+        resource_sets: Dict[str, Optional[str]] = {}
+        resource_set_instances: List["Instance"] = (
+            types["std::ResourceSet"].get_all_instances() if types and "std::ResourceSet" in types else []
+        )
+        assert resource_mapping is not None
```

Comment:
I would move this to the start of the method. 

Perhaps also add an assert that types is not None